src/core/dynamic_system.jl

```diff
@@ -1,60 +1,95 @@
+
 """
-    sesolve(H::AbstractOperator, ψ_0::Ket, t_range::StepRangeLen; e_ops::Vector{AbstractOperator}=(AbstractOperator)[])
-    sesolve(H::Function, ψ_0::Ket, t_range::StepRangeLen; e_ops::Vector{Operator}=(Operator)[])
+    ShrodingerProblem is a structure that is defined to solve the shrodinger equation in time-domain using sesolve(). 
 
-Solves the Shrodinger equation:
+**Fields**
+- `H` -- a function that retrurns the  Hamiltonian operator (of any subtype of `AbstractOperator`) as a function of time.
+- `init_state` -- initital state (`Ket`) of a quantum system
+- `tspan` -- time interval for which the system has to be simulated. 
+        For instance: 
+            tspan=(0.0,1.0) evaluates the output from t=0.0 to t=1.0
+- `e_ops` -- list of operators for which the expected value 
+    (the observables) will be evaluated at each time step in t_range. 
 
-``\\frac{d \\Psi}{d t}=-i \\hat{H}\\Psi``
+"""
+Base.@kwdef struct ShrodingerProblem{T<:AbstractOperator, S<:Complex}
+    H::Function
+    init_state::Ket{S}
+    tspan::Tuple{Float64,Float64}
+    e_ops::Vector{T}
+end
 
-and returns the final state Ket, and a Vector of observables evaluated at each time step. 
+"""
+    A LindbladProblem is a structure that is defined to solve the Lindblad master equation in time-domain using mesolve(). 
 
 **Fields**
-- `H` -- the Hamiltonian operator (of any subtype of `AbstractOperator`) or a 
-        function that returns the Hamiltonian as a function of time.
-- `ψ_0` -- initital state (`Ket`) of a quantum system
-- `t_range` -- time interval for which the system has to be simulated. 
+- `H` -- a function that retrurns the  Hamiltonian operator (of any subtype of `AbstractOperator`) as a function of time.
+- `init_state` -- initital state density matrix (`DenseOperator`) of a quantum system
+- `tspan` -- time interval for which the system has to be simulated. 
         For instance: 
-            t_range=0:10 evaluates the output using time 
-            steps: 0,1,2,...,10. 
-            t_range=0:0.01:1 evaluates the output using 
-            time steps: 0,0.01,0.02,...,1.0 
-
-- `e_ops` -- list of operators for which the expected value 
+            tspan=(0.0,1.0) evaluates the output from t=0.0 to t=1.0
+- `e_ops` -- list of operators (type DenseOperator) for which the expected value 
```

Comment:
Why should the `e_ops` be dense? Wouldn't this be slow compared to a more specialized operator?

Reply:
The problem is that for the time being we have not implemented mixed type algebra. Once we do that, we can remove this requirement.

Reply:
I've added an issue, https://github.com/anyonlabs/Snowflake.jl/issues/239.